### DISCUSSION TITLE
Possible to do Count and Update/Create in single transaction?

### DISCUSSION THREAD

**Discussion by taylorhlabs** [prisma/prisma#5647] [Q&A] [answered]
In the following example, I get the count of existing activity and then (using a transaction) create a new activity record and update the activity count in the related resource. I have 2 questions:
1. Is it possible to do this in fewer function calls? ie Count the Activities for the Resource and update the Count column in 1 call?
2. I assume this requires 2 trips to the database; one to get the current count and one to run the transaction. a) Is there a way to do this in a single trip to the database? b) Am I correct in assuming that a transaction guarantees a single trip to the database to run all statements in the transaction?

```ts
// await get count
let resultActivityCount = await prisma.activity.count({
  where: { resourceId: resourceId }
});
resultActivityCount++;

// update in single transaction
const newActivity = prisma.activity.create({
  data: {
    user: {
      connect: { id: token.uid }
    },
    resource: {
      connect: { id: resourceId }
    },
    text: activityText,
  }
});

const updateResource = prisma.resource.update({
  where: { id: resourceId },
  data: { activityCount: resultActivityCount }
});

const [ resultActivity, resultResource ] = await prisma.$transaction([ newActivity, updateResource ]);

```

**Comment by ryands17** [comment, marked as answer]
@taylorhlabs 👋 

1. In this case as you need the count, you would need 2 calls, one for fetching the count and the other for the transaction.
2. Yes you're correct in assuming this. 2 trips will be required and the transaction will be in a single trip.

As for completing this in a single query, one option could be a raw query where you update the value by fetching the count in a subquery in the update call. Let me know if that helps :)

**Reply by taylorhlabs** [reply]
Thanks @ryands17  for the input on this. I got it down to a single transaction using `queryRaw` in case anyone else finds it useful. It requires more queries but only requires a single trip to the database.
```ts
    const createComment = prisma.activity.create({
      data: {
        user: {
          connect: { id: token.uid }
        },
        resource: {
          connect: { id: resourceId }
        },
        text: commentText,
        type: commentType
      }
    });

    const updateResource = prisma.$executeRaw(`
      UPDATE resources
      SET activity_count = (SELECT COUNT(activities.id) FROM activities WHERE resource_id = ${resourceId})
      WHERE id = ${resourceId};
    `);

    const selectResource = prisma.resource.findUnique({
      where: { id: resourceId }
    });

    const [resultCreateComment, resultUpdateResource, resultSelectResource] = await prisma.$transaction([
      createComment,
      updateResource,
      selectResource
    ]);
    // return new comment and updated resource with activity count
    res.json({ resultCreateComment, resultSelectResource });
```

**Comment by Rahmo** [comment]
hi  @ryands17 
how would I do this count this in this form within single trip 
```ts
        const navigations = await this.prisma.feeds.findMany({
            where,
            skip:  skip * (take),
            take:  take,
            }
        )
        const count = await this.prisma.feeds.count({ where })
```
im doing two calls here would be great to have that within one trip

**Reply by ryands17** [reply]
@Rahmo this is not possible in a single trip. Although you can use a transaction to make it atomic:

```ts
const [navigations, count] = await this.prisma.$transaction([
  this.prisma.feeds.findMany({
            where,
            skip:  skip * (take),
            take:  take,
            }
        ),
      this.prisma.feeds.count({ where })
])
```

**Comment by alexey-sh** [comment]
```
transaction:
  table.count
  // log count
  table.create
```

3 simultaneous requests logs 0, 1, 1 in any isolationLevel

So I think it's not possible, at least with the prisma

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
